### PR TITLE
feat(irs-api):[TRI-API] Adding helm chart for ESR Spike

### PR DESCRIPTION
**Pull request by ds-ext-ooyeneye**

### CODE REVIEW

File: chart/aaswrapper/values-esr.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        - path: /
+          pathType: Prefix
+          backend:
+            service:
+              name: aaswrapper-esr
+              port:
+                number: 9191
+  tls:
+    - hosts:
+        - "irs-aas-wrapper-esr.int.demo.catena-x.net"
```
Comment (ds-jhartmann) [inline chart/aaswrapper/values-esr.yaml:27]: Why is the url on int cluster here? Insn't it supposed to be on dev?

File: chart/edc-consumer-controlplane/values-esr.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        - path: /api/v1/ids
+          pathType: Prefix
+          backend:
+            service:
+              name: edc-consumer-controlplane-esr
+              port:
+                number: 8282
+  tls:
+    - hosts:
+        - "irs-consumer-controlplane-esr.int.demo.catena-x.net"
```
Comment (ds-jhartmann) [inline chart/edc-consumer-controlplane/values-esr.yaml:26]: Why is the url on int cluster here? Insn't it supposed to be on dev?

File: chart/irs/values-esr.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+ingress:
+  hosts:
+    - host: "irs-esr.int.demo.catena-x.net"
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls:
+    - hosts:
+        - "irs-esr.int.demo.catena-x.net"
```
Comment (ds-jhartmann) [inline chart/irs/values-esr.yaml:14]: Why is the url on int cluster here? Insn't it supposed to be on dev?

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: chart/edc-consumer-controlplane/values-esr.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+## Ingress declaration to expose the network service.
+ingress:
+  enabled: true
+  className: "nginx"
+  annotations:
+    nginx.ingress.kubernetes.io/ssl-passthrough: "false"
+    nginx.ingress.kubernetes.io/backend-protocol: "HTTP"
+    nginx.ingress.kubernetes.io/force-ssl-redirect: "false"
+  hosts:
+    - host: "irs-consumer-controlplane-esr.int.demo.catena-x.net"
```
Comment (ds-ext-kmassalski) [inline chart/edc-consumer-controlplane/values-esr.yaml:15]: We agreed to deploy on DEV

File: chart/irs/values-esr.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+springprofile: esr
+
+image:
+  tag: main
+
+ingress:
+  hosts:
+    - host: "irs-esr.int.demo.catena-x.net"
```
Comment (ds-ext-kmassalski) [inline chart/irs/values-esr.yaml:8]: We agreed to deploy on DEV

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]
Missing components. There are more things to deploy, not only aaswrapper, edc controlplane, dataplane and irs.

Besides of that mixed configuration, sometimes there is DEV, sometimes INT.

File: chart/edc-consumer-controlplane/values-esr.yaml (hunk 2 above)
Comment (ds-ext-ooyeneye) [reply chart/edc-consumer-controlplane/values-esr.yaml]: Done

File: chart/edc-consumer-dataplane/values-esr.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    edc.oauth.private.key.alias=irs-int-daps-key
+    edc.oauth.provider.audience=idsc:IDS_CONNECTORS_ALL
+    edc.oauth.provider.jwks.url=https://daps1.int.demo.catena-x.net/.well-known/jwks.json
+    edc.oauth.public.key.alias=irs-int-daps-cert
+    edc.oauth.token.url=https://daps1.int.demo.catena-x.net/token
+    edc.receiver.http.endpoint=http://aaswrapper-esr:9192/callback
+    edc.vault.clientid=c1778e7d-db16-4404-9080-26d2280d2f63
+    edc.vault.clientsecret=<path:traceability-irs/data/int/controlplane#vaultClientSecret>
+    edc.vault.name=cxtsi-dev001-vault
+    edc.vault.tenantid=495463c3-0991-4659-9cc5-94b4a3f7b1d6
```
Comment (ds-jhartmann) [inline chart/edc-consumer-dataplane/values-esr.yaml:45]: Azure vault is no longer available. You need to use the hashicorp vault images

File: chart/edc-consumer-controlplane/values-esr.yaml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    edc.oauth.public.key.alias=irs-int-daps-cert
+    edc.oauth.token.url=https://daps1.int.demo.catena-x.net/token
+    edc.receiver.http.endpoint=http://aaswrapper-esr:9192/callback/endpoint-data-reference
+    edc.transfer.proxy.endpoint=https://irs-consumer-dataplane-esr.dev.demo.catena-x.net/api/public
+    edc.transfer.proxy.token.signer.privatekey.alias=azure-vault-token-signer-private-key
+    edc.transfer.proxy.token.verifier.publickey.alias=azure-vault-token-signer-private-key-pub
+    edc.vault.clientid=c1778e7d-db16-4404-9080-26d2280d2f63
+    edc.vault.clientsecret=<path:traceability-irs/data/dev/controlplane#vaultClientSecret>
+    edc.vault.name=cxtsi-dev001-vault
+    edc.vault.tenantid=495463c3-0991-4659-9cc5-94b4a3f7b1d6
```
Comment (ds-jhartmann) [inline chart/edc-consumer-controlplane/values-esr.yaml:51]: Azure vault is no longer available. You need to use the hashicorp vault images

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

**Review by ds-ext-kmassalski — CHANGES_REQUESTED** [review]
Not passing acceptence criteria - read task description and comment from Sprint planning 2 one more time.
Not deployed entire env, missing postman collection, different naming then agreed in task.

File: chart/edc-consumer-controlplane/values-esr.yaml (hunk 7 above)
Comment (ds-ext-ooyeneye) [reply chart/edc-consumer-controlplane/values-esr.yaml]: Had been change to hashicorp vault

File: chart/irs/values-esr.yaml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+keycloak:
+  oauth2:
+    clientId: <path:traceability-irs/data/int/keycloak/oauth2#clientId>
+    clientSecret: <path:traceability-irs/data/int/keycloak/oauth2#clientSecret>
+    clientTokenUri: <path:traceability-irs/data/int/keycloak/oauth2#tokenUri>
+    jwkSetUri: <path:traceability-irs/data/int/keycloak/oauth2#jwkSetUri>
+
+aasProxy:
+  submodel:
```
[the remark this answers is not in the document]
Comment (ds-ext-ooyeneye) [reply chart/irs/values-esr.yaml:26]: Adapted

**Comment by ds-ext-ooyeneye** [comment]
Please see TRI-666 for continual reviewing of this branch, issue mentioned here had been resolved in TRI-666

**Comment by ds-jkreutzfeld** [comment]
Is this binary file needed? [chart/edc-postgresql/charts/common-1.16.1.tgz](https://github.com/catenax-ng/product-item-relationship-service/pull/257/files#diff-7b306faf6e96e16b43e7e1f69a4ec26239bdc62b2a56e99e9836991f9eddd730)

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**